docs/spec.md

```diff
@@ -6559,6 +6558,68 @@ tensor. Depending on the element type, does the following:
 
+## Dialect Interop
+
+At the moment, StableHLO programs in the wild sometimes contain operations that
+are not defined by StableHLO.
+
+### Module, Function, Call and Return
+
+StableHLO uses upstream MLIR operations for ModuleOp, FuncOp, CallOp, and
+ReturnOp. This was done for better interop with existing MLIR machinery, as many
+useful passes are written targeting FuncOp and ModuleOp, and many compilation
+pipelines expect these ops to be present.  Full compatibility guarantees are
+applied to these ops. If anything ever changes about these ops in an
+incompatible way (i.e. removal), StableHLO equivalents will be added to preserve
+compatibility.
+
+### CHLO
+
+The CHLO opset contains higher level operations that decompose to StableHLO.
+Currently there are no compatibility guarantees for CHLO. For compatibility
+guarantees, the [CHLO-to-StableHLO decomposition pass](https://github.com/openxla/stablehlo/blob/12fd0a9e7b3c6f3dea3defc513870c962e62726d/stablehlo/transforms/Passes.td#L119)
+must be used prior to serialization.
+
+### Shape Operations
+
+It is a common use case in the community to use certain operations from core
+MLIR dialects in dynamic StableHLO programs to perform shape computations.
+Most commonly, these include [`shape` dialect](https://mlir.llvm.org/docs/Dialects/ShapeDialect/)
+ops like `shape_of` or `num_elements`, [`tensor` dialect](https://mlir.llvm.org/docs/Dialects/TensorOps/)
+ops like `dim` or `from_elements`, and the builtin `index` type.
+
+The [Dynamism RFC > O2](https://github.com/openxla/stablehlo/blob/main/rfcs/20230704-dynamism-101.md#o2)
+denotes these as out of scope, however some support for `index` types is
+included for interop purposes. There are no compatibility guarantees for these
+ops or types. The [Shape-to-StableHLO](https://github.com/openxla/stablehlo/blob/12fd0a9e7b3c6f3dea3defc513870c962e62726d/stablehlo/transforms/Passes.td#L136)
+pass can be used to convert these operations to fully supported StableHLO ops.
+
+## Deprecated Operations
+
+There are several StableHLO operations that were inherited from MHLO which are
+deprecated and on the way out of StableHLO. The full details on these removals
+can be found in the [StableHLO v1.0 Cleanup #2283](https://github.com/openxla/stablehlo/pull/2283).
+
+These operations fall into a few categories:
+
+* "Not in HLO" category of StableHLO operations - they were initially part of
+  the StableHLO opset but have been later deemed to not fit it well:
+  `broadcast`, `create_token`, `cross-replica-sum`, `dot`, `einsum`,
+  `torch_index_select`, `unary_einsum`
+  ([#3](https://github.com/openxla/stablehlo/issues/3)).
+* Unused ops - These operations may have been useful at some point, but the ops
+  were either underdeveloped, or the pipelines using these ops have been
+  refactored to not require them anymore. This includes `map`, `tuple` and
+  `get_tuple_element`, `rng`.
+
+Some of these ops can confidently be easily removed (`broadcast`,
```

Comment:
Nit: "can confidently be easily removed" sounds a little weird -- consider rephrasing and maybe providing additional context: "can be removed easily given that they can be expressed using existing ops"